composer/core/time.py

```diff
@@ -554,13 +566,13 @@ class Timestamp(NamedTuple):
         :class:`Timestamp` should not be instantiated directly; instead use :meth:`Timer.get_timestamp`.
 
     Attributes:
-        epoch (Time[int]): epoch at a snapshot.
-        batch (Time[int]): batch at a snapshot.
-        batch_in_epoch (Time[int]): The number of batches seen in the epoch at a snapshot.
-        sample (Time[int]): sample at a snapshot.
-        sample_in_epoch (Time[int]): The number of samples seen in the epoch at a snapshot.
-        token (Time[int]): token at a snapshot.
-        token_in_epoch (Time[int]): The number of tokens seen in the epoch at a snapshot.
+        epoch (Time[int]): The epoch.
+        batch (Time[int]): The batch.
+        batch_in_epoch (Time[int]): The number of batch seen in the last epoch.
+        sample (Time[int]): The sample count.
+        sample_in_epoch (Time[int]): The number of samples seen in the last epoch.
+        token (Time[int]): The token count.
+        token_in_epoch (Time[int]): The number of tokens seen in the last epoch.
```

Comment:
All of the `x_in_epoch` attributes are incorrect, assuming `Timestamp` could be generated in the middle of the epoch. They are supposed to refer to the number of batches/samples/tokens seen so far in the current epoch, at the time of the snapshot.

Also, could we make `epoch`/`batch`/`sample`/`token`'s definitions consistent with each other?